src/node.c

```diff
@@ -23,20 +23,21 @@ static CMARK_INLINE bool S_is_inline(cmark_node *node) {
 }
 
 static bool S_can_contain(cmark_node *node, cmark_node *child) {
-  cmark_node *cur;
-
-  if (node == NULL || child == NULL) {
+  if (node == NULL || child == NULL || node == child) {
     return false;
   }
 
-  // Verify that child is not an ancestor of node or equal to node.
```

Comment:
I can see why, if child has no children, it cannot be an ancestor of node.
But what is replacing the check to verify that child is not equal to node?

Reply:
The equality test is now a few lines above.